internal/metrics/remote_write.go

```diff
@@ -153,19 +160,58 @@ func (r *RemoteWrite) Update(config models.DeviceConfigurationMessage) error {
 			return fmt.Errorf("metrics remote write configuration is invalid. RequestNumSamples has to greater than 0")
 		}
 
+		// create TLS client config
+		clientConfig := config_util.HTTPClientConfig{}
+
+		if newConfig.CaCert != "" {
+			// create new file for CA. can't use same file cause client watches for changes in the file we pass to it.
+			caFile, err := ioutil.TempFile(r.dataDir, ServerCAFileNamePattern)
+			if err != nil {
+				log.Error("could not create temp file for metics remote write server CA")
+				return err
+			}
+
+			caFilePath := caFile.Name()
+
+			_, err = caFile.WriteString(newConfig.CaCert)
+			if err == nil {
+				err = caFile.Sync()
+				if err == nil {
+					err = caFile.Close()
+				}
+			}
+			if err != nil {
+				_ = caFile.Close()
+				log.Errorf("could not write to metics remote write server CA file %s", caFilePath)
+				return err
+			}
+
+			clientConfig.TLSConfig = config_util.TLSConfig{
+				CAFile: caFilePath,
+			}
+		}
+
+		// create client
 		client, err := remote.NewWriteClient(r.deviceID, &remote.ClientConfig{
 			Timeout: model.Duration(newConfig.TimeoutSeconds * int64(time.Second)),
 			URL: &config_util.URL{
 				URL: serverURL,
 			},
+			HTTPClientConfig: clientConfig,
 		})
 		if err != nil {
+			_ = os.Remove(clientConfig.TLSConfig.CAFile)
```

Comment:
Is it ok to not check `err` here?

Reply:
I am already in error state. There's nothing I can do with error returned from Remove.